src/blockly/generators/propc/base.js

```diff
@@ -59,172 +59,124 @@ Blockly.Blocks.math_number = {
+                    .appendField(this.lastBlockText, 'TITLE');
+        }
+        this.setWarningText(warnText);
+    },
+    getRangeArray: function () {
+        var range = ['N', -100, 100, Number(this.getFieldValue('NUM'))];
+
+        if (this.outputConnection && this.outputConnection.targetBlock()) {
+            var connectingBlock = this.outputConnection.targetBlock()
+            var fieldListing = connectingBlock.getInputWithBlock(this).getRange();
+            if (fieldListing) {
+                var rangeVals = fieldListing.split(',');
+                range[0] = rangeVals[0];
+                if (rangeVals[0] === 'S' || rangeVals[0] === 'R') {
+                    range[1] = Number(rangeVals[1]);
+                    range[2] = Number(rangeVals[2]);
+                } else if (rangeVals[0] === 'A') {
+                    for (var idx = 4; idx < rangeVals.length - 3; idx++) {
+                        range[idx] = Number(rangeVals[idx - 3]);
                     }
-                    this.connection_id_ = null;
-                    rangeVals = ['N', '-100', '100', '0'];
                 }
             }
-            range[2] = Number(this.getFieldValue('NUM'));
-            if (rangeVals) {
-                if (rangeVals[0] === 'R') {
-                    if (range[2] < range[0]) {
-                        this.setWarningText('WARNING: Your value is too small!  It must be greater than or equal to ' + range[0].toString(10));
-                    } else if (range[2] > range[1]) {
-                        this.setWarningText('WARNING: Your value is too large!  It must be less than or equal to ' + range[1].toString(10));
-                    } else {
-                        this.setWarningText(null);
-                    }
-                } else if (rangeVals[0] === 'A') {
-                    var warnMsg = 'none';
-                    for (let idx = 0; idx < range.length; idx++)
-                        if (range[2] === Number(rangeVals[idx]))
-                            warnMsg = 'match';
-                    if (warnMsg === 'none') {
-                        this.setWarningText('WARNING: The value you entered is not available or not allowed!');
-                    } else {
-                        this.setWarningText(null);
-                    }
-                } else if (rangeVals[0] === 'S') {
-                    this.setWarningText(null);
+        }
+        return range;
+    },
+    updateShape: function (rangeArray) {
+        var fieldType = 'number';
+        var blockText = '';
+        var warnText = null;
+        var range = rangeArray || this.getRangeArray();
+        var fieldToAdd = new Blockly.FieldNumber(range[3].toString(10), null, null, 1);
+
+        if (range[0] === 'R') {
+            if (range[3] < range[1]) {
+                warnText = 'WARNING: Your value is too small!  It must be greater than or equal to ' + range[1].toString(10);
```

Comment:
We usually place all message strings in the _message.js file. Should the strings in warnText be handled the same way?

Reply:
Yup - but as a seperate issue.  If internationalization becomes a need/priority, we'll need to do it for ALL user-facing strings in the code.

Reply:
Still, we could apply it here since we're modifying this code anyway, right?

Reply:
Sure - Give me a few minutes.